insights/parsers/freeipa_healthcheck_log.py

```diff
@@ -0,0 +1,69 @@
+"""
+freeipa_healthcheck_log - File ``/var/log/ipa/healthcheck/healthcheck.log``
+===========================================================================
+
+This module provides plugins access to file ``/var/log/ipa/healthcheck/healthcheck.log``
+
+This file contains a list of items, each item representing a check made on the local IPA server.
+The file can be either single-line or multi-line indented.
+
+Typical content of file ``/var/log/ipa/healthcheck/healthcheck.log`` is::
+
+    [
+      {
+        "source": "ipahealthcheck.meta.services",
+        "check": "certmonger",
+        "result": "SUCCESS",
+        "uuid": "ceeae285-6a98-498c-89f9-38cc60279e28",
+        "when": "20190719155725Z",
+        "duration": "0.014892",
+        "kw": {
+          "status": true
+        }
+      },
+      {
+        "source": "ipahealthcheck.meta.services",
+        "check": "dirsrv",
+        "result": "SUCCESS",
+        "uuid": "b5faf965-2df9-4b3d-b6d3-0b29bbf293a5",
+        "when": "20190719155725Z",
+        "duration": "0.013207",
+        "kw": {
+          "status": true
+        }
+      }
+    ]
+
+The list of errors can be accessed via the errors property.
+Examples:
+
+    >>> from insights.parsers.freeipa_healthcheck_log import FreeIPAHealthCheckLog
+    >>> from insights.tests import context_wrap
+    >>>
+    >>> LONG_FREEIPA_HEALTHCHECK_LOG_FAILURES = '''
+    ... [{"source": "ipahealthcheck.system.filesystemspace", "check": "FileSystemSpaceCheck",
+    ... "severity": 2, "uuid": "ceadd564-f42f-4640-a3ae-b67f1aa29dc4", "when": "20190711141807Z",
+    ... "duration": null, "kw": {"msg": "/var/log/: free space under threshold: 400 MiB < 1024 MiB",
+    ... "store": "/var/log/", "free_space": 400, "threshold": 1024}}]
+    ... '''.strip()
+    >>> FREEIPA_HEALTHCHECK_LOG_FAILURES = "".join(LONG_FREEIPA_HEALTHCHECK_LOG_FAILURES.splitlines())
+    >>>
+    >>> log_obj = FreeIPAHealthCheckLog(context_wrap(FREEIPA_HEALTHCHECK_LOG_FAILURES))
+    >>> for error in log_obj.errors:
+    ...     assert error['check'] == 'FileSystemSpaceCheck'
+    ...     assert error['source'] == 'ipahealthcheck.system.filesystemspace'
```

Comment:
These examples should use the `healthcheck` object that you create in your tests on line 66 instead of creating a new object here.  `healthcheck` will be set to whatever test value you use on line 66, and can just be used directly in the examples here.  Generally it is good to use test data that is the same as the "typical contents of the file" shown above the Examples section so that a user can follow the examples.

Reply:
hi @bfahr thanks for the input, I believe I've solved that as well in the latest push.